Crash "Fatal error while checking" with metaclass

### Bug description

When parsing the following file:

```python
"""Fatal error while checking."""
class Meta(type(object), type(object)):
    pass

class Class(metaclass=Meta):
    pass

```

pylint crashed with a ``AstroidError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\lint\pylinter.py", line 811, in _lint_file
    check_astroid_module(module)
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\lint\pylinter.py", line 1085, in check_astroid_module
    retval = self._check_astroid_module(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\lint\pylinter.py", line 1135, in _check_astroid_module
    walker.walk(node)
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\utils\ast_walker.py", line 94, in walk
    self.walk(child)
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\utils\ast_walker.py", line 91, in walk
    callback(astroid)
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\checkers\typecheck.py", line 1040, in visit_classdef
    if _is_invalid_metaclass(metaclass):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\checkers\typecheck.py", line 748, in _is_invalid_metaclass
    mro = metaclass.mro()
          ^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3066, in mro
    return self._compute_mro(context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3054, in _compute_mro
    unmerged_mro = list(clean_duplicates_mro(unmerged_mro, self, context))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 155, in clean_duplicates_mro
    raise DuplicateBasesError(
astroid.exceptions.DuplicateBasesError: Duplicates found in MROs (Meta), (type, object), (type, object), (type, type) for <ClassDef.Meta l.2 at 0x26ca8902bd0>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\lint\pylinter.py", line 775, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\lint\pylinter.py", line 813, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```


### Configuration

_No response_

### Command used

```shell
pylint bug.py
```


### Pylint output

```shell
************* Module bug
bug.py:2:0: C0115: Missing class docstring (missing-class-docstring)
bug.py:2:0: E0241: Duplicate bases for class 'Meta' (duplicate-bases)
bug.py:5:0: C0115: Missing class docstring (missing-class-docstring)
Exception on node <ClassDef.Class l.5 at 0x26ca8903410> in file 'C:\Code\Bug\bug.py'
Traceback (most recent call last):
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\utils\ast_walker.py", line 91, in walk
    callback(astroid)
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\checkers\typecheck.py", line 1040, in visit_classdef
    if _is_invalid_metaclass(metaclass):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\pylint\checkers\typecheck.py", line 748, in _is_invalid_metaclass
    mro = metaclass.mro()
          ^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3066, in mro
    return self._compute_mro(context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3054, in _compute_mro
    unmerged_mro = list(clean_duplicates_mro(unmerged_mro, self, context))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\bers\.pyenv-win-venv\envs\project_3.11\Lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 155, in clean_duplicates_mro
    raise DuplicateBasesError(
astroid.exceptions.DuplicateBasesError: Duplicates found in MROs (Meta), (type, object), (type, object), (type, type) for <ClassDef.Meta l.2 at 0x26ca8902bd0>.
bug.py:1:0: F0002: bug.py: Fatal error while checking 'bug.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in 'C:\Users\bers\AppData\Local\pylint\pylint\Cache\pylint-crash-2023-05-18-09-05-35.txt'. (astroid-error)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 7.14/10, -7.14)
```


### Expected behavior

I have no idea if the code examples makes any sense, but `pylint` should not crash.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)]
```


### OS / Environment

Windows 10 21H2

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8698.bugfix
+++ b/doc/whatsnew/fragments/8698.bugfix
@@ -1,0 +1,3 @@
+Fix crash in ``invalid-metaclass`` check when a metaclass had duplicate bases.
+
+Closes #8698

--- a/pylint/checkers/typecheck.py
+++ b/pylint/checkers/typecheck.py
@@ -747,7 +747,10 @@ def _no_context_variadic(
 
 
 def _is_invalid_metaclass(metaclass: nodes.ClassDef) -> bool:
-    mro = metaclass.mro()
+    try:
+        mro = metaclass.mro()
+    except (astroid.DuplicateBasesError, astroid.InconsistentMroError):
+        return True
     return not any(is_builtin_object(cls) and cls.name == "type" for cls in mro)
 
 

--- a/tests/functional/i/invalid/invalid_metaclass.py
+++ b/tests/functional/i/invalid/invalid_metaclass.py
@@ -5,6 +5,8 @@
 # pylint: disable=multiple-statements
 
 import abc
+from pathlib import Path
+from typing import Protocol
 
 import six
 from unknown import Unknown
@@ -67,4 +69,26 @@ class Invalid(metaclass=invalid_metaclass_1):  # [invalid-metaclass]
 
 
 class InvalidSecond(metaclass=invalid_metaclass_2):  # [invalid-metaclass]
+    pass
+
+
+class MetaclassWithInvalidMRO(type(object), type(object)):  # [duplicate-bases]
+    pass
+
+
+class FifthInvalid(metaclass=MetaclassWithInvalidMRO):  # [invalid-metaclass]
+    pass
+
+
+class Proto(Protocol):
+    ...
+
+
+class MetaclassWithInconsistentMRO(type(Path), type(Proto)):  # [inconsistent-mro]
+    pass
+
+
+class SixthInvalid(  # [invalid-metaclass]
+    Path, Proto, metaclass=MetaclassWithInconsistentMRO
+):
     pass

--- a/tests/functional/i/invalid/invalid_metaclass.txt
+++ b/tests/functional/i/invalid/invalid_metaclass.txt
@@ -1,6 +1,10 @@
-invalid-metaclass:41:0:41:18:FirstInvalid:Invalid metaclass 'int' used:UNDEFINED
-invalid-metaclass:45:0:45:19:SecondInvalid:Invalid metaclass 'InvalidAsMetaclass' used:UNDEFINED
-invalid-metaclass:49:0:49:18:ThirdInvalid:Invalid metaclass '2' used:UNDEFINED
-invalid-metaclass:53:0:53:19:FourthInvalid:Invalid metaclass 'Instance of invalid_metaclass.InvalidAsMetaclass' used:UNDEFINED
-invalid-metaclass:65:0:65:13:Invalid:Invalid metaclass 'int' used:UNDEFINED
-invalid-metaclass:69:0:69:19:InvalidSecond:Invalid metaclass '1' used:UNDEFINED
+invalid-metaclass:43:0:43:18:FirstInvalid:Invalid metaclass 'int' used:UNDEFINED
+invalid-metaclass:47:0:47:19:SecondInvalid:Invalid metaclass 'InvalidAsMetaclass' used:UNDEFINED
+invalid-metaclass:51:0:51:18:ThirdInvalid:Invalid metaclass '2' used:UNDEFINED
+invalid-metaclass:55:0:55:19:FourthInvalid:Invalid metaclass 'Instance of invalid_metaclass.InvalidAsMetaclass' used:UNDEFINED
+invalid-metaclass:67:0:67:13:Invalid:Invalid metaclass 'int' used:UNDEFINED
+invalid-metaclass:71:0:71:19:InvalidSecond:Invalid metaclass '1' used:UNDEFINED
+duplicate-bases:75:0:75:29:MetaclassWithInvalidMRO:Duplicate bases for class 'MetaclassWithInvalidMRO':UNDEFINED
+invalid-metaclass:79:0:79:18:FifthInvalid:Invalid metaclass 'MetaclassWithInvalidMRO' used:UNDEFINED
+inconsistent-mro:87:0:87:34:MetaclassWithInconsistentMRO:Inconsistent method resolution order for class 'MetaclassWithInconsistentMRO':UNDEFINED
+invalid-metaclass:91:0:91:18:SixthInvalid:Invalid metaclass 'MetaclassWithInconsistentMRO' used:UNDEFINED

